src/lib-components/SectionPagination.vue

```diff
@@ -1,116 +1,189 @@
 <script lang="ts" setup>
-import { computed } from 'vue'
-
-// SVGs
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
+import type { Ref } from 'vue'
+import { useWindowSize } from '@vueuse/core'
 import SvgIconArrowRight from 'ucla-library-design-tokens/assets/svgs/icon-arrow-right.svg'
+import { useTheme } from '@/composables/useTheme'
 
 // COMPONENTS
 import SmartLink from '@/lib-components/SmartLink.vue'
 
 // PROPS & DATA
-const { nextTo, previousTo } = defineProps({
+const { nextTo, previousTo, pages, initialCurrentPage } = defineProps({
   nextTo: {
     type: String,
-    default: '',
+    required: false,
   },
   previousTo: {
     type: String,
-    default: '',
+    required: false,
+  },
+  pages: {
+    type: Number,
+    required: false,
+  },
+  initialCurrentPage: {
+    type: Number,
+    required: false,
   },
 })
+const emit = defineEmits(['changePage']) // let parent component know when page changes
+const theme = useTheme()
+const maxPages = ref(10) // default # of buttons that will fit in container, gets recalculated onMount & resize
+const leftPages = ref([33]) // an array of numbers representing the page buttons that will appear ( we start with a single '33' so we can measure the width of a button to calc maxPages)
+const currPage = ref(1) // current page, defaults to 1
+const pageButtons: Ref<HTMLElement | null> = ref(null)
+
+// METHODS
+function handlePageChange(item: number) {
+  if (initialCurrentPage && pages) {
+    if (currPage.value !== item) {
+      currPage.value = item
+      generateLeftPages()
+      emit('changePage', item)
+    }
+  }
+}
+function generateLeftPages() {
```

Comment:
In my testing this did not perform the same as the current function. The current function keeps all the buttons on one line during screen resize. This does not, I think because some of the logic was stripped out to recalculate # of buttons when the truncation buttons are there vs when they are not. 

Please test yourself and let me know if you get different results, and I can retry.